.githooks/pre-commit

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+# Redirect output to stderr.
+exec 1>&2
+
+output=$(git clang-format --diff --staged)
+if [[ "$output" != "" && "$output" != "no modified files to format" ]]; then
+  echo Code formatting changed some files, please review.
+  exit 1
+fi
```

Comment:
Does it work when there is not installed clang-format?

Reply:
No. But I guess one shouldn't install the pre-commit hook then.